skimage/util/dtype.py

```diff
@@ -508,3 +509,27 @@ def img_as_bool(image, force_copy=False):
 
     """
     return convert(image, np.bool_, force_copy)
+
+
+def convert_to_float(image, preserve_range):
```

Comment:
Could you explain why you moved this function? Why include it in this PR?

Reply:
It caused a circular import when i tried to import _shared.warnings

Reply:
I think eventually we won't use our own `warn` function, and just import warnings directly. I'll probably revert this.

I think jni will want to move things around, but we will have to find a more systematic way to do things.